src/server/game/Spells/SpellEffects.cpp

```diff
@@ -4474,17 +4474,25 @@ void Spell::EffectInebriate(SpellEffIndex /*effIndex*/)
     }
 
     uint8 currentDrunk = player->GetDrunkValue();
-    uint8 drunkMod = damage;
-    if (currentDrunk + drunkMod > 100)
-    {
+    int32 drunkMod = damage;
+
+    if (drunkMod == 0)
+        return;
+
+    // drunkMod may contain values ​​that are guaranteed to cause uint8 overflow/underflow (examples: 29690, 46874)
+    // In addition, we would not want currentDrunk to become more than 100.
+    // So before adding the values, let's check that everything is fine.
+    if (drunkMod > 0 && drunkMod > static_cast<int32>(100 - currentDrunk))
```

Comment:
I would prefer braces (...) around the conditions for readability here, also on the following if (...) conditions.
Would you agree? 🙂

Reply:
> I would prefer braces (...) around the conditions for readability here, also on the following if (...) conditions. Would you agree? 🙂

I don't think that extra braces will improve code readability in this case. I prefer to explicitly use optional braces just to avoid confusion with priority of `&&` and `||` operators. But that's a different case.

Reply:
Alright, I'd still prefer them but it's not my call as I'm not maintainer.